File: backend/LendQ.Core/Entities/PaymentAllocation.cs
namespace LendQ.Core.Entities;

public class PaymentAllocation : BaseEntity
{
    public Guid PaymentTransactionId { get; set; }
    public PaymentTransaction PaymentTransaction { get; set; } = null!;

    public Guid InstallmentId { get; set; }
    public ScheduleInstallment Installment { get; set; } = null!;

    public decimal Amount { get; set; }
}

File: backend/app/models/payment_allocation.py
import uuid
from datetime import datetime, timezone

from app.extensions import db


class PaymentAllocation(db.Model):
    __tablename__ = "payment_allocations"

    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    transaction_id = db.Column(
        db.String(36), db.ForeignKey("payment_transactions.id"), nullable=False, index=True
    )
    installment_id = db.Column(
        db.String(36), db.ForeignKey("schedule_installments.id"), nullable=False
    )
    amount = db.Column(db.Numeric(12, 2), nullable=False)
    created_at = db.Column(db.DateTime, nullable=False, default=lambda: datetime.now(timezone.utc))

    transaction = db.relationship("PaymentTransaction", back_populates="allocations")
    installment = db.relationship("ScheduleInstallment", foreign_keys=[installment_id])
